### PR TITLE
lncli: allow PSBT to be read from file

### LINKED ISSUE #5080: PSBT channel funding fails with very long PSBT strings
### Background

Please see [this reddit post](https://old.reddit.com/r/lightningnetwork/comments/lyd7dk/psbt_channel_funding_issues/). It appears that this user can use the PSBT channel funding support just fine except when using very big PSBTs, where they get this message: `[lncli] base64 decode failed: illegal base64 data at input byte 4092` 

### Your environment
Unknown

### Steps to reproduce

Try to open a channel using the `--psbt` functionality. It allegedly succeeds with small PSBT strings but fails with long PSBT strings.

### Expected behaviour

Channel funding should work fine with any length PSBT string.

### Actual behaviour

Allegedly channel funding does not work with very long PSBT strings.

### Preliminary investigation:

The error message is from `cmd/lncli/cmd_open_channel.go`, line 521 or line 721.  I suspect the root cause is in `readLine`, where `fmt.Scan` is called only once in the goroutine.  I am not a golang programmer so take this hunch with a grain of salt :).

### PR DESCRIPTION
**Pull request by guggero**
Fixes #5080.

The N_TTY_BUF_SIZE kernel parameter dictates how many characters can be
pasted into a terminal window. This cannot be circumvented by reading
the input in a different manner.
To avoid the problem fully, we instead allso allow the user to type in a
path to a text file that is read if it exists. That way the PSBT can be
as long as needed.

### CODE REVIEW

**Comment by Crypt-iQ** [comment]
What would a reasonable maximum size for a psbt be?

**Review by wpaulino — APPROVED** [review]

**Comment by guggero** [comment]
> What would a reasonable maximum size for a psbt be?

That's hard to say. The raw transaction itself can be up to 400k and still be valid. The psbt for it can have a lot more data added to it though (additional information for each input and output, custom data) so it's very hard to set a limit.

Sure, we can cap it at 1MB just to be sure the user doesn't run into an out of memory situation. But other than that I don't think we can add a sane size check.

**Comment by Crypt-iQ** [comment]
> > What would a reasonable maximum size for a psbt be?
> 
> That's hard to say. The raw transaction itself can be up to 400k and still be valid. The psbt for it can have a lot more data added to it though (additional information for each input and output, custom data) so it's very hard to set a limit.
> 
> Sure, we can cap it at 1MB just to be sure the user doesn't run into an out of memory situation. But other than that I don't think we can add a sane size check.

A cap of 1MB seems like a good idea

File: cmd/lncli/cmd_open_channel.go [hunk 1]
```diff
@@ -635,6 +646,60 @@ func printChanPending(update *lnrpc.OpenStatusUpdate_ChanPending) error {
...
+// prevents pasting more than 4096 characters (on most systems) into a terminal.
+func readTerminalOrFile(quit chan struct{}) (string, error) {
+	maybeFile, err := readLine(quit)
+	if err != nil {
+		return "", err
+	}
+
+	// Absolute file paths normally can't be longer than 255 characters so
+	// we don't even check if it's a file in that case.
+	if len(maybeFile) > 255 {
```
Comment (Crypt-iQ) [inline cmd/lncli/cmd_open_channel.go:663]: it's possible for a psbt to be <= 255 right?

**Review by Crypt-iQ — APPROVED** [review]
LGTM

File: cmd/lncli/cmd_open_channel.go (hunk 1 above)
Comment (guggero) [reply cmd/lncli/cmd_open_channel.go:663]: Yes. That's why we return the string directly if a file with that name doesn't exist.

**Pull request merged**